collectors/plugins.d/pluginsd_parser.c

```diff
@@ -1849,6 +1919,76 @@ static void virt_fnc_got_data_cb(BUFFER *wb, int code, void *callback_data)
 }
 
 #define VIRT_FNC_TIMEOUT 1
+void call_virtual_function_async(BUFFER *wb, RRDHOST *host, const char *name, const char *payload, rrd_call_function_async_callback callback, void *callback_data) {
+    PARSER *parser = NULL;
+
+    //TODO simplify (as we really need only first parameter to get plugin name maybe we can avoid parsing all)
+    char *words[PLUGINSD_MAX_WORDS];
+    char *function_with_params = strdupz(name);
+    BUFFER *function_out = buffer_create(strlen(name), NULL);
```

Comment:
May can defer buffer creation to so if there is an error, you can avoid the buffer free